### PR TITLE
Remove proxy from `FeedRouter`

### PR DESCRIPTION
**Pull request by martinkersner**
# Description

* Function to remove (`removeProxyBulk`) proxy address from `FeedRouter`
* Keep track of registered feeds (`feedNames`) in `FeedRouter`

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.

## Deployment

- [ ] Should publish npm package
- [ ] Should publish Docker image

## Summary by CodeRabbit

- **New Features**
  - Enhanced feed management in the application with the ability to add, update, and remove proxy addresses for different feed names.
  - Introduced a new array, `feedNames`, for better organization and retrieval of feed information.

- **Refactor**
  - Improved event management by replacing a single event with more specific events for proxy address additions, removals, and updates.

- **Tests**
  - Added comprehensive test cases to validate the functionality of adding, updating, and removing proxy addresses, ensuring robust feed management.

### CODE REVIEW

File: contracts/v0.2/src/FeedRouter.sol [hunk 1]
```diff
@@ -115,53 +126,101 @@ contract FeedRouter is Ownable, IFeedRouter {
...
+	    }
+	}
+
+	if (!found) {
+	    feedNames.push(_feedName);
+	    emit ProxyAddressAdded(_feedName, _proxyAddress);
+	} else {
+	    emit ProxyAddressUpdated(_feedName, _proxyAddress);
+	}
+    }
```
[the remark this answers is not in the document]
Comment (martinkersner) [reply contracts/v0.2/src/FeedRouter.sol]: fixed at https://github.com/Bisonai/orakl/pull/1408/commits/047fa2cb417687a69868aa5ac64fb054fb264f99

**Comment by martinkersner** [comment]
@coderabbitai review

File: contracts/v0.2/src/FeedRouter.sol [hunk 2]
```diff
@@ -16,15 +16,17 @@ import {IFeedProxy} from "./interfaces/IFeedProxy.sol";
  * can find all supported tokens at https://config.orakl.network.
  */
 contract FeedRouter is Ownable, IFeedRouter {
-    mapping(string => address) public feedProxies;
+    mapping(string => address) public feedToProxies;
```
Comment (nick-bisonai) [inline contracts/v0.2/src/FeedRouter.sol:19]: Just curious, is there a specific reason why you thought `feedToProxies` is better than `feedProxies`?

**Review by nick-bisonai — APPROVED** [review]
lgtm!

File: contracts/v0.2/src/FeedRouter.sol (hunk 2 above)
Comment (martinkersner) [reply contracts/v0.2/src/FeedRouter.sol:19]: No, specific reason. At first I was using different variable names (`feeds` -> `feedNames`) and the name was too similar to `feedProxies` so I renamed it, but now it has no problem so I can move it back.

Comment (nick-bisonai) [reply contracts/v0.2/src/FeedRouter.sol:19]: I don't mind :) thank you for the explanation

**Pull request merged**